src/mod_muc_room.erl

```diff
@@ -4334,6 +4334,9 @@ tab_add_online_user(JID, StateData) ->
     US = {LUser, LServer},
     Room = StateData#state.room,
     Host = StateData#state.host,
+    ServerHost = StateData#state.server_host,
+    MucJID = StateData#state.jid,
+    ejabberd_hooks:run(join_room, ServerHost, [ServerHost, Room, Host, JID, MucJID]),
```

Comment:
The function is called `tab_add_online_user/2`. So, it should update `muc_online_users` table and that's it.
A hook call can be done outside of the function, for example

```erlang
tab_add_online_user(JID, StateData),
run_join_room_hook(JID, StateData),
```

Than you need to put hook running logic inside a new function `run_join_room_hook`.

Reply:
So would you want me to add run_join_room_hook(JID, StateData) inside of this function?
https://github.com/esl/MongooseIM/blob/1bc8515f9c7f070ce157523a70910bef06f8b91c/src/mod_muc_room.erl#L1654 Then make a separate method for calling the hook?